provider/cmd/app/app.go

```diff
@@ -349,6 +368,11 @@ func Run(ctx context.Context, opts Options) error {
 	}
 
 	g, ctx := errgroup.WithContext(ctx)
+
+	g.Go(func() error {
```

Comment:
Should this function still be run in a goroutine if the metrics server is disabled?

Reply:
go routines will end and program still is running. I don't see any big problem with that and i have code in one function.